scripts/metrics/metrics.py

```diff
@@ -87,65 +90,64 @@
     if (n_hvgs is not None):
         if (adata_int.n_vars < n_hvgs):
             raise ValueError("There are less genes in the corrected adata than specified for HVG selection")
-       
+
     # check input files
     if adata.n_obs != adata_int.n_obs:
         message = "The datasets have different numbers of cells before and after integration."
         message += "Please make sure that both datasets match."
         raise ValueError(message)
-    
-    #check if the obsnames were changed and rename them in that case
+
+    # check if the obsnames were changed and rename them in that case
     if len(set(adata.obs_names).difference(set(adata_int.obs_names))) > 0:
-        #rename adata_int.obs[batch_key] labels by overwriting them with the pre-integration labels
+        # rename adata_int.obs[batch_key] labels by overwriting them with the pre-integration labels
         new_obs_names = ['-'.join(idx.split('-')[:-1]) for idx in adata_int.obs_names]
 
         if len(set(adata.obs_names).difference(set(new_obs_names))) == 0:
             adata_int.obs_names = new_obs_names
         else:
             raise ValueError('obs_names changed after integration!')
-            
-    #batch_key might be overwritten, so we match it to the pre-integrated labels
+
+    # batch_key might be overwritten, so we match it to the pre-integrated labels
     adata_int.obs[batch_key] = adata_int.obs[batch_key].astype('category')
-    if not np.array_equal(adata.obs[batch_key].cat.categories,adata_int.obs[batch_key].cat.categories):
```

Comment:
@LuckyMD I changed this line, as batch categories were the same, but the order of the batches was not. The check assumes that the indices are the same for both integrated and nonintegrated datasets, which was the case for my test data. Ideally we'd need a fix for this issue in the integration wrappers themselves, but that's a separate problem for scib

Reply:
I thought I left a comment on this part as well. Are you sure that the next line doesn't automatically match indices in the Series assignment and therefore not mind ordering differences? I thought I checked this at the time.

Reply:
The old code breaks, as category names are consistent, but the data is still mislabeled. So the category check does not recognise that these label assignments are mismatched

```
# unintegrated
# print(adata.obs[batch_key].value_counts())
3    564
4    559
0    551
1    550
2    506

# integrated
Name: batch, dtype: int64
0    564
1    559
4    551
3    550
2    506
```

When using `if not np.array_equal(adata.obs[batch_key], adata_int.obs[batch_key]):`  instead to also check the complete array (rather than the unique set of categories) the relabeled dataset seems consistent
```
# unintegrated
# print(adata.obs[batch_key].value_counts())

3    564
4    559
0    551
1    550
2    506
Name: batch, dtype: int64

# integrated & renamed
print(adata_int.obs[batch_key].value_counts())

3    564
4    559
0    551
1    550
2    506
Name: batch, dtype: int64
```

Reply:
We need to fix this batch ordering in the `metrics.py` script, no? Otherwise scanorama and mnn won't run through.

Reply:
We need to check value counts similarity.

Reply:
and confirm that assignments of pd.Series match by indices if there is a re-ordering